Vanishing Point Analysis

taokinbo opened the pull request:
This PR uses Canny edge detection and Hough Line to detect vanishing points within the photo.

Currently, our analysis:

- [ ] Detects and extends significant (nonvertical and nonhorizontal) lines

- [ ] Returns the coordinate of the average point of a group of coordinates with the most intersections or None if no coordinate was found

TODO

- [ ] Tune parameters of Hough lines and Canny to filter out insignificant lines and find more accurate lines

@ryaanahmed @clarkce

ryaanahmed commented on backend/app/analysis/find_vanishing_point.py:
```diff
@@ -0,0 +1,220 @@
+                filter_lines.append(line)
+        except ZeroDivisionError:
+            pass
+    # van_point = find_van_coord(filter_lines, image.shape[0], image.shape[1])
```
Here's an example of what I'm talking about in the review message -- all of this is commented out and really should be deleted, as it makes the routine difficult to read.

ryaanahmed reviewed (changes requested):
The new algorithm looks like it gives better results! Great work there.

Since initial review over in #65, there's now a lot of unused code that's still here but just commented out.

Please either remove these commented out lines if they're actually no longer useful, or in the case of something like the original `find_van_coord` (where it's actually useful to have the original implementation hanging around, as we might want to repurpose parts of it), please comment why it's still there.

taokinbo commented:
@ryaanahmed 
I have removed all the unnecessary comments and changed our tester to fit the format of the other tester functions. There shouldn't be any merging issues now.

ryaanahmed commented:
This work rolled into #85, so closing this PR, and merging the work in over there.
